lib/TcpConnection.js

```diff
@@ -142,6 +168,21 @@ module.exports = function factory (userOptions) {
       .removeAllListeners('error')
   }
 
+  function recoverEnqueuedData () {
+    if (recoveryQueue.isEmpty()) {
+      return
+    }
+    const item = recoveryQueue.peek()
+    socket.write(item.data, item.encoding, (err) => {
+      if (err) {
```

Comment:
we should emit some event in this case

Reply:
We could but currently we do not emit any event when we are unable to write data (the first time), maybe we should? What is your opinion on using https://nodejs.org/api/process.html#processemitwarningwarning-options? In my opinion, it's better than printing the error using `process.stderr.write`.

Reply:
> // TODO we must propagate the events from the socket to the outputStream

That's an interesting one but I guess we should use "non-standard" event names, no? I mean, if the socket is closed, we shouldn't emit a `close` event on the outputStream (Writable stream), right? So we would probably use a `socketClose` event name?

Reply:
I've added the following warning:

```js
emitWarning('unable to recover data because an error occurred while writing data to the TCP socket', { detail: err.message })
```

Do you think we should emit an event named `socketError` instead? Do you think we should also emit this event here:

https://github.com/pinojs/pino-socket/blob/989b395a46f5d8729817515be9fb0a9dcd5686b5/lib/TcpConnection.js#L60

Reply:
We should not be using `emitWarning` for this, but emitting an event here and using that event to print something to stdout on the main cli file